l3kernel/l3str-format.dtx

```diff
@@ -90,7 +87,7 @@
 %
 % \section{Formatting various data-types}
 %
-% \begin{function}[EXP]{\tl_format:Nn, \tl_format:cn, \tl_format:nn}
+% \begin{function}[EXP, added = 2025-06-09]{\tl_format:Nn, \tl_format:cn, \tl_format:nn}
```

Comment:
Perhaps the docs for `\<type>_format:..` functions can be moved to the corresponding `l3<type>` module, like we did in
- #1622

For example, the doc for `\tl_format:(N|n)n` can be moved to `l3tl` module, under the subsection "Using token lists" (or under a new subsection titled "Formatting token lists").

The implementation looks centric, thus can be left in `l3str-format.dtx` and this module would be another implementation-only module.